#   1. Importar librerías

In [214]:
from glob import glob
import os
import pandas as pd
import numpy as np

# 2. Acceder a los datos de 'FELICIDAD'

In [215]:
# Verificar el directorio de trabajo
os.getcwd()

'c:\\Users\\fkva_\\OneDrive\\Escritorio\\LIDE\\08_Python\\python_tareas\\tarea_2'

In [216]:
# Determinar documentos terminados en .csv
glob('*/*/*/*/*.csv')

['datos-de-miercoles\\datos\\2019\\2019-04-17\\cambio_lealtades.csv',
 'datos-de-miercoles\\datos\\2019\\2019-04-17\\personajes_libro.csv',
 'datos-de-miercoles\\datos\\2019\\2019-04-17\\tiempo_pantalla.csv',
 'datos-de-miercoles\\datos\\2019\\2019-04-24\\gapminder_es.csv',
 'datos-de-miercoles\\datos\\2019\\2019-05-01\\comercio_hispanoamerica_mundo_agregado.csv',
 'datos-de-miercoles\\datos\\2019\\2019-05-08\\datos_uip.csv',
 'datos-de-miercoles\\datos\\2019\\2019-05-22\\tuberculosis_oms.csv',
 'datos-de-miercoles\\datos\\2019\\2019-05-29\\mes.csv',
 'datos-de-miercoles\\datos\\2019\\2019-05-29\\terremotos.csv',
 'datos-de-miercoles\\datos\\2019\\2019-06-05\\buenos_aires.csv',
 'datos-de-miercoles\\datos\\2019\\2019-06-05\\cdmx.csv',
 'datos-de-miercoles\\datos\\2019\\2019-06-05\\rio.csv',
 'datos-de-miercoles\\datos\\2019\\2019-06-05\\santiago.csv',
 'datos-de-miercoles\\datos\\2019\\2019-06-12\\vinos.csv',
 'datos-de-miercoles\\datos\\2019\\2019-06-19\\cambios.csv',
 'datos-de-mierc

In [217]:
# Determinar el archivo que contiene los datos de 'FELICIDAD'
datos = glob('*/*/*/*/*.csv')

In [218]:
# Leer los datos de 'FELICIDAD'
datos_felicidad = pd.read_csv(datos[24])
datos_felicidad

pais  anio  escalera_vida   log_pib  soporte_social  \
0     Afghanistán  2008       3.723590  7.168690        0.450662   
1     Afghanistán  2009       4.401778  7.333790        0.552308   
2     Afghanistán  2010       4.758381  7.386629        0.539075   
3     Afghanistán  2011       3.831719  7.415019        0.521104   
4     Afghanistán  2012       3.782938  7.517126        0.520637   
...           ...   ...            ...       ...             ...   
1699     Zimbabue  2014       4.184451  7.562753        0.765839   
1700     Zimbabue  2015       3.703191  7.556052        0.735800   
1701     Zimbabue  2016       3.735400  7.538829        0.768425   
1702     Zimbabue  2017       3.638300  7.549491        0.754147   
1703     Zimbabue  2018       3.616480  7.553395        0.775388   

      expectativa_vida  libertad  generosidad  percepcion_corrupcion  \
0            50.799999  0.718114     0.177889               0.881686   
1            51.200001  0.678896     0.200178               0.850035   
2            51.599998  0.600127     0.134353               0.706766   
3            51.919998  0.495901     0.172137               0.731109   
4            52.240002  0.530935     0.244273               0.775620   
...                ...       ...          ...                    ...   
1699         52.380001  0.642034    -0.048634               0.820217   
1700         53.799999  0.667193    -0.097354               0.810457   
1701         54.400002  0.732971    -0.068105               0.723612   
1702         55.000000  0.752826    -0.069670               0.751208   
1703         55.599998  0.762675    -0.038384               0.844209   

      afecto_positivo  afecto_negativo  confianza  calidad_democracia  \
0            0.517637         0.258195   0.612072           -1.929690   
1            0.583926         0.237092   0.611545           -2.044093   
2            0.618265         0.275324   0.299357           -1.991810   
3            0.611387         0.267175   0.307386           -1.919018   
4            0.710385         0.267919   0.435440           -1.842996   
...               ...              ...        ...                 ...   
1699         0.725214         0.239111   0.566209           -0.985267   
1700         0.715079         0.178861   0.590012           -0.893078   
1701         0.737636         0.208555   0.699344           -0.900649   
1702         0.806428         0.224051   0.682647           -0.988153   
1703         0.710119         0.211726   0.550508                 NaN   

      calidad_entrega  de_escalera_pais_anio  gini_banco_mundial  \
0           -1.655084               1.774662                 NaN   
1           -1.635025               1.722688                 NaN   
2           -1.617176               1.878622                 NaN   
3           -1.616221               1.785360                 NaN   
4           -1.404078               1.798283                 NaN   
...               ...                    ...                 ...   
1699        -1.484067               2.079248                 NaN   
1700        -1.357514               2.198865                 NaN   
1701        -1.374650               2.776363                 NaN   
1702        -1.350867               2.656848                 NaN   
1703              NaN               2.498696                 NaN   

      gini_banco_mundial_promedio  
0                             NaN  
1                             NaN  
2                             NaN  
3                             NaN  
4                             NaN  
...                           ...  
1699                        0.432  
1700                        0.432  
1701                        0.432  
1702                        0.432  
1703                        0.432  

[1704 rows x 17 columns]

# 3. Análisis

## 3.1. Medida de tendencia central y dispersión de la felicidad a nivel mundial para el primer y último año

In [219]:
# Tabla felicidad filtrada con primer y último año
felicidad_filtrada_anio = datos_felicidad[(datos_felicidad['anio'] == datos_felicidad['anio'].min()) | (datos_felicidad['anio'] == datos_felicidad['anio'].max())]
felicidad_filtrada_anio

pais  anio  escalera_vida   log_pib  soporte_social  \
10    Afghanistán  2018       2.694303  7.494588        0.507516   
21        Albania  2018       5.004403  9.412399        0.683592   
28        Algeria  2018       5.043086  9.557952        0.798651   
45      Argentina  2018       5.792797  9.809972        0.899912   
58        Armenia  2018       5.062449  9.119424        0.814449   
...           ...   ...            ...       ...             ...   
1654    Venezuela  2018       5.005663  9.270281        0.886882   
1667      Vietnam  2018       5.295547  8.783416        0.831945   
1678        Yemen  2018       3.057514       NaN        0.789422   
1690       Zambia  2018       4.041488  8.223958        0.717720   
1703     Zimbabue  2018       3.616480  7.553395        0.775388   

      expectativa_vida  libertad  generosidad  percepcion_corrupcion  \
10           52.599998  0.373536    -0.084888               0.927606   
21           68.699997  0.824212     0.005385               0.899129   
28           65.900002  0.583381    -0.172413               0.758704   
45           68.800003  0.845895    -0.206937               0.855255   
58           66.900002  0.807644    -0.149109               0.676826   
...                ...       ...          ...                    ...   
1654         66.500000  0.610855    -0.176156               0.827560   
1667         67.900002  0.909260    -0.039124               0.808423   
1678         56.700001  0.552726          NaN               0.792587   
1690         55.299999  0.790626     0.036644               0.810731   
1703         55.599998  0.762675    -0.038384               0.844209   

      afecto_positivo  afecto_negativo  confianza  calidad_democracia  \
10           0.424125         0.404904   0.364666                 NaN   
21           0.713300         0.318997   0.435338                 NaN   
28           0.591043         0.292946        NaN                 NaN   
45           0.820310         0.320502   0.261352                 NaN   
58           0.581488         0.454840   0.670828                 NaN   
...               ...              ...        ...                 ...   
1654         0.759221         0.373658   0.260700                 NaN   
1667         0.692222         0.191061        NaN                 NaN   
1678         0.461114         0.314870   0.308151                 NaN   
1690         0.702698         0.350963   0.606715                 NaN   
1703         0.710119         0.211726   0.550508                 NaN   

      calidad_entrega  de_escalera_pais_anio  gini_banco_mundial  \
10                NaN               1.408344                 NaN   
21                NaN               2.640531                 NaN   
28                NaN               1.973943                 NaN   
45                NaN               2.472559                 NaN   
58                NaN               2.102111                 NaN   
...               ...                    ...                 ...   
1654              NaN               2.891930                 NaN   
1667              NaN               1.676448                 NaN   
1678              NaN               2.402008                 NaN   
1690              NaN               2.783419                 NaN   
1703              NaN               2.498696                 NaN   

      gini_banco_mundial_promedio  
10                            NaN  
21                       0.303250  
28                       0.276000  
45                       0.460938  
58                       0.319250  
...                           ...  
1654                     0.497167  
1667                     0.362750  
1678                     0.357000  
1690                     0.527400  
1703                     0.432000  

[163 rows x 17 columns]

### Medida de tendencia central (promedio)

In [220]:
# Cálculo de felicidad promedio del primer y último año
felicidad_promedio = felicidad_filtrada.groupby('anio')['escalera_vida'].mean()
felicidad_promedio 

anio
2005    7.358502
2018    7.275893
Name: escalera_vida, dtype: float64

#### Conclusión
Según los datos obtenidos, para el año 2005 el índice de felicidad promedio se situó en 6.446164; mientras que, para el año 2018 el mismo índice se situó en 5.502134. Estos valores sugieren una reducción en este indicador para estos años de análisis.

### Medida de dispersión (desviación estándar)

In [221]:
# Cálculo de desviación estándar de la felicidad del primer y último año
felicidad_std = felicidad_filtrada.groupby('anio')['escalera_vida'].std()
felicidad_std 

anio
2005    0.025192
2018    0.139865
Name: escalera_vida, dtype: float64

#### Conclusión
Según los datos obtenidos, para el año 2005 la desviación estándar del índice de felicidad se situó en 0.919143; mientras que, para el año 2018 el mismo índice se situó en 1.103461. Estos valores sugieren un aumento en la variabilidad de la felicidad, en otras palabras, las mediciones de felicidad son más dispersas en 2018 en comparación con el año 2005.

## 3.2. Comparación de valores a nivel mundial con dos países de su elección en el primer año reportado y el último año reportado

### Análisis 1: Comparación de valores sin determinar medidas de tendencia central ni dispersión

In [222]:
# Determinar países de interés
paises = ['Ecuador', 'Suecia']

In [223]:
# Utilizar la tabla filtrada por año en el inciso anterior para filtrar también por los países de interés
felicidad_filtrada_pais = felicidad_filtrada_anio[felicidad_filtrada_anio['pais'].isin(paises)]
felicidad_filtrada_pais

pais  anio  escalera_vida    log_pib  soporte_social  \
439   Ecuador  2018       6.128010   9.274455        0.851345   
1432   Suecia  2005       7.376316  10.626856        0.951470   
1444   Suecia  2018       7.374792  10.766932        0.930680   

      expectativa_vida  libertad  generosidad  percepcion_corrupcion  \
439          68.500000  0.869364    -0.104021               0.830743   
1432         71.199997  0.964395          NaN                    NaN   
1444         72.599998  0.941725     0.069573               0.262797   

      afecto_positivo  afecto_negativo  confianza  calidad_democracia  \
439          0.876475         0.328171   0.402085                 NaN   
1432         0.839870         0.150766   0.440849             1.44275   
1444         0.822676         0.160755   0.494396                 NaN   

      calidad_entrega  de_escalera_pais_anio  gini_banco_mundial  \
439               NaN               2.485788                 NaN   
1432          1.81479               1.618058               0.268   
1444              NaN               1.601522                 NaN   

      gini_banco_mundial_promedio  
439                      0.496000  
1432                     0.274154  
1444                     0.274154

In [224]:
# Calcular el promedio de 'felicidad' mundial
promedio_mundial = datos_felicidad['escalera_vida'].mean()
promedio_mundial

5.4371550357397735

### Análisis 2: Comparación de valores determinando medidas de tendencia central y dispersión

#### Medida de tendencia central (promedio)

In [225]:
# Calcular el promedio de 'felicidad' de cada país
promedio_pais = datos_felicidad.groupby('pais')['escalera_vida'].mean()
promedio_pais

pais
Afghanistán    3.705495
Albania        4.990211
Alemania       6.792368
Algeria        5.481872
Angola         4.420299
                 ...   
Venezuela      6.203183
Vietnam        5.293443
Yemen          3.886234
Zambia         4.631722
Zimbabue       4.029770
Name: escalera_vida, Length: 164, dtype: float64

In [226]:
# Calcular el promedio de 'felicidad' mundial
promedio_mundial = datos_felicidad['escalera_vida'].mean()
promedio_mundial

5.4371550357397735

In [227]:
# Determinar países de interés
pais_1 = "Ecuador"
pais_2 = "Suecia"

In [228]:
# Seleccionar información de países de interés
promedio_paises = promedio_pais.loc[[pais_1, pais_2]]
promedio_paises

pais
Ecuador    5.764949
Suecia     7.371575
Name: escalera_vida, dtype: float64

In [229]:
print(f"Promedio de 'felicidad' para : {promedio_paises}")
print(f"Promedio de 'felicidad' mundial: {promedio_mundial}")

Promedio de 'felicidad' para : pais
Ecuador    5.764949
Suecia     7.371575
Name: escalera_vida, dtype: float64
Promedio de 'felicidad' mundial: 5.4371550357397735


##### Conclusión
 Según los datos obtenidos, se concluye que Suecia tiene el mayor promedio de felicidad (7.371575), seguido por Ecuador (5.764949) y el promedio mundial (5.437155036). Por lo tanto, según estos datos, Suecia tiene el mayor nivel de felicidad.

#### Medida de dispersión (desviación estándar)

In [230]:
# Calcular la desviación estándar de 'felicidad' de cada país
std_pais = datos_felicidad.groupby('pais')['escalera_vida'].std()
std_pais

pais
Afghanistán    0.664635
Albania        0.470496
Alemania       0.228729
Algeria        0.422567
Angola         0.815301
                 ...   
Venezuela      1.018408
Vietnam        0.217101
Yemen          0.590778
Zambia         0.480803
Zimbabue       0.595815
Name: escalera_vida, Length: 164, dtype: float64

In [231]:
# Calcular la desviación estándar de 'felicidad' mundial
std_mundial = datos_felicidad['escalera_vida'].std()
std_mundial

1.1211490772800061

In [232]:
# Determinar países de interés
pais_1 = "Ecuador"
pais_2 = "Suecia"

In [233]:
# Seleccionar información de países de interés
std_paises = std_pais.loc[[pais_1, pais_2]]
std_paises

pais
Ecuador    0.394485
Suecia     0.106308
Name: escalera_vida, dtype: float64

In [234]:
print(f"Promedio de 'felicidad' para : {std_paises}")
print(f"Promedio de 'felicidad' mundial: {std_mundial}")

Promedio de 'felicidad' para : pais
Ecuador    0.394485
Suecia     0.106308
Name: escalera_vida, dtype: float64
Promedio de 'felicidad' mundial: 1.1211490772800061


##### Conclusión
 Según los datos obtenidos, la desviación estándar más baja se encuentra en Suecia (0.106308), seguida por Ecuador (0.394485) y el valor más alto es para el mundo en general (1.121149077). Por lo tanto, se puede concluir que, en términos de homogeneidad en los niveles de felicidad, Suecia parece ser el país más feliz, seguido por Ecuador y luego el mundo en general.